scripts/copy_from_upstream/.CMake/alg_support.cmake/add_enable_by_alg.fragment

```diff
@@ -16,11 +16,11 @@ endif()
 {% if platform['operating_systems'] %}if(CMAKE_SYSTEM_NAME MATCHES "{{ platform['operating_systems']|join('|') }}")
 {% endif -%}
 if((OQS_DIST_ARM64_V8_BUILD OR (OQS_USE_ARM_NEON_INSTRUCTIONS{% for flag in platform['required_flags'] -%} {%- if not loop.last or loop.first %} AND {% endif -%}OQS_USE_ARM_{{ flag|upper }}_INSTRUCTIONS {%- endfor -%})))
-{% if family['name'] == "kyber" and impl['upstream']['name'] == 'pqclean' -%}
-if(((CMAKE_C_COMPILER_ID STREQUAL "GNU") AND (CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL "9.4.0")) OR ((CMAKE_CXX_COMPILER_ID STREQUAL "GNU") AND (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "9.4.0")))
+{% if impl['upstream']['name'] == 'pqclean' -%}
```

Comment:
Is this change to enable (or disable) _all_ PQClean algs as per rather complex build environment rules intentional? This not only impacts Saber.

Reply:
Yes it is. Since we don't advertise a minimum compiler version, Douglas suggested adding a compiler version check for the minimum version of gcc required to get access to the intrinsics we need. The modification in this PR allows (apple)clang to using the aarch64 optimized algorithms.

The reason the if statement doesn't check for "kyber" anymore is because these intrinsics appear in saber as well, so I made it more general.

Reply:
Understood. The question was whether this (same check) applies _to all_ algorithms imported from PQClean (not just kyber and saber)? But then again, we only have aarch64 optimized code for those algorithms, so it's fine. Thanks for the explanation.